tensorflow_addons/losses/general_loss.py

```diff
@@ -0,0 +1,309 @@
+        alpha=1: Charbonnier/pseudo-Huber loss.
+        alpha=2: L2 loss.
+      scale: The `scale` parameter of the loss. When |y_true - y_pred| < scale,
+        the loss is an L2-like quadratic bowl, and when
+        |y_true - y_pred| > scale the loss function takes on a different shape
+        according to `alpha`. Must be a tensorflow tensor or numpy array of
+        single-precision floats. Default value for `scale` is 1.
+      approximate: a bool, where if True, this function returns an approximate
+        and faster form of the loss, as described in the appendix of the paper.
+        This approximation holds well everywhere except as residual `x` and
+        `alpha` approach zero. By default, `approximate` is set to False.
+      epsilon: A float that determines how inaccurate the "approximate" version
+        of the loss will be. Larger values are less accurate but more
+        numerically stable. Must be greater than single-precision machine
+        epsilon. Default value of `epsilon` is 1e-6.
+
+    Returns:
+      The losses for each element of residual `x`, in the same shape as `x`.
+      This is returned as a TensorFlow graph node of single precision floats.
+
+    Raises:
+      ValueError: If `epsilon` is less than or equal to single-precision
+        machine epsilon. And if `scale` is less than or equal to 0.
+      TypeError: If `alpha` or `scale` is of different dtype than `y_true`
+        and `y_pred`.
+    """
+
+    @typechecked
+    def __init__(
+        self,
+        from_logits: bool = False,
+        alpha: FloatTensorLike = 2.0,
+        scale: FloatTensorLike = 1.0,
+        approximate: bool = False,
+        epsilon: FloatTensorLike = 1e-6,
+        reduction: str = tf.keras.losses.Reduction.NONE,
+        name: str = "general_loss",
+    ):
+        super().__init__(name=name, reduction=reduction)
+        self.from_logits = from_logits
+        self.alpha = alpha
+        self.scale = scale
+        self.approximate = approximate
+        self.epsilon = epsilon
+
+    def call(self, y_true, y_pred):
+        return general_loss(
+            y_true,
+            y_pred,
+            alpha=self.alpha,
+            scale=self.scale,
+            approximate=self.approximate,
+            epsilon=self.epsilon,
+            from_logits=self.from_logits,
+        )
+
+    def get_config(self):
+        config = {
+            "from_logits": self.from_logits,
+            "alpha": self.alpha,
+            "scale": self.scale,
+            "approximate": self.approximate,
+            "epsilon": self.epsilon,
+        }
+        base_config = super().get_config()
+        return {**base_config, **config}
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+@tf.function
+def general_loss(
+    y_true: TensorLike,
+    y_pred: TensorLike,
+    alpha: FloatTensorLike = tf.constant(2.0, dtype=tf.dtypes.float64),
+    scale: FloatTensorLike = tf.constant(1.0, dtype=tf.dtypes.float64),
```

Comment:
No, there is no such reason. We have to pick one of them. I picked float64. We can change it to float32. Should I?

Reply:
To make the signature simpler, I suggest using python scalar instead, and converting them to tensors in the function. It will allow users to use scalars and tensors.

Reply:
Done as per the comment by @gabrieldemarmiesse .